sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala

```diff
@@ -320,4 +320,11 @@ class HiveOrcSourceSuite extends OrcSuite with TestHiveSingleton {
       }
     }
   }
+
+  test("SPARK-31580: Read a file written before ORC-569") {
+    assume(HiveUtils.isHive23) // Hive 1.2 doesn't use Apache ORC
```

Comment:
I haven't seen this before, that's a cool feature :)
That being said I'm not sure about the version number check here, looking at the related JIRAs https://issues.apache.org/jira/browse/HIVE-22480 & https://issues.apache.org/jira/browse/ORC-569 it seems like it should be fixed earlier? Or is that JIRA wrong?

Reply:
The Hive JIRA is correct, but all versions in HIVE-22480 are not released officially.
> 1.3.0, 1.2.3, 2.1.2, 2.2.1

From Hive 2.3.x, Hive starts to use Apache ORC. During the transition, Hive side fix seems to be not migrated to Apache ORC correctly.

Reply:
Thanks for the additional context :) Sounds reasonable.